README.md

```diff
@@ -115,6 +115,19 @@ A full list of all the unit tests we run in our CI can be found [here](.github/s
 MMTk is also tested with the VM bindings we are maintaining by running standard test/benchmark suites for the VMs.
 For details, please refer to each VM binding repository.
 
+## Supported Rust versions
+
+As a research project, MMTk is quite aggressive in following the latest releases of the Rust
+toolchain. The minimum supported Rust version (MSRV) is "N-1", that is, one minor version before the
```

Comment:
I think that's also OK.  My understanding is,

1.  We always guarantee that "N-1" is supported.
2.  But the actual `rust-version` field in `Cargo.toml` *may* sometimes be older than `N-1` if we didn't make use of any newer features during the last 6 weeks of development, and no dependencies require a newer `rust-version`.
3.  If, during our development, we made a PR that uses a newer feature introduced in `N-m` where `M>=1`, we have a good reason to bump `rust-version` to `N-m`.
4.  The CI shall test with both the latest stable version and the `rust-version` in `Cargo.toml`.

That still sounds good, but I still prefer bumping the MSRV eagerly (even for "no reason"), for the following reasons:
1.  If the MSRV is already bumped after a release, we don't have to deliberately bump it during development.
2.  If we cannot guarantee that `N-2` is supported, our users still cannot rely on `N-2` even if the `rust-version` in `Cargo.toml` is sometimes `N-2` or older.  That's because it may change in the next release.  Although the user may lock the version of `mmtk-core` to a version to satisfy their MSRV requirement, we should not encouraged this (at least for now) because MMTk is being rapidly developed, and relying on an outdated MMTk now will make it difficult for the users to migrate to the latest MMTk in the future.

However, we may change the policy if MMTk reached a stable state.  By that time, we may prefer letting the MSRV "linger" at a version, like you suggested, if we do not make use of newer features.  That will allow the users to use a newer `mmtk-core` while still using an older Rust toolchain.  At that time, it should be reasonable for the user to lock the version of `mmtk-core` because `mmtk-core` is relatively stable.  But I don't think MMTk has reached such a state.